sphinx_js/typedoc.py

```diff
@@ -194,7 +194,10 @@ def _convert_node(self, node) -> Tuple[TopLevel, List[dict]]:
             # many attr of Functions.
             sigs = node.get('signatures')
             first_sig = sigs[0]  # Should always have at least one
-            first_sig['sources'] = node['sources']
+            parent = node
+            while 'sources' not in node and '__parent' in node:
+                parent = node['__parent']
```

Comment:
Whoops, sorry, I suggested an infinite loop.  Try this:

```suggestion
            while 'sources' not in parent and '__parent' in parent:
                parent = parent['__parent']
```